src/mlpack/methods/ann/init_rules/glorot_init.hpp

```diff
@@ -77,8 +77,9 @@ class GlorotInitializationType
    *
    * @param W Weight matrix to initialize.
    */
-  template<typename eT>
-  void Initialize(arma::Mat<eT>& W);
+  template<typename MatType>
+  void Initialize(MatType& W,
+      typename std::enable_if_t<IsMatrix<MatType>::value>* = 0);
```

Comment:
```suggestion
      const typename std::enable_if_t<IsMatrix<MatType>::value>* = 0);
```
Not sure if this is really needed, but it's what we do through the rest of the code and it can't hurt...